api/src/main/java/io/strimzi/api/kafka/model/template/KafkaClusterTemplate.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.template;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+
+import java.io.Serializable;
+
+/**
+ * Representation of a template for Kafka cluster resources.
+ */
+@Buildable(
+        editableEnabled = false,
+        generateBuilderPackage = false,
+        builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "statefulset", "pods", "service", "headlessService"})
+public class KafkaClusterTemplate implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private StatefulSetTemplate statefulset;
+    private PodTemplate pods;
+    private ServiceTemplate service;
+    private ServiceTemplate headlessService;
+
+    @Description("Template for Kafka stateful set.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public StatefulSetTemplate getStatefulset() {
+        return statefulset;
+    }
+
+    public void setStatefulset(StatefulSetTemplate statefulset) {
+        this.statefulset = statefulset;
+    }
+
+    @Description("Template for Kafka pods.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public PodTemplate getPods() {
+        return pods;
+    }
+
+    public void setPods(PodTemplate pods) {
+        this.pods = pods;
+    }
+
+    @Description("Template for Kafka boostrap service.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public ServiceTemplate getService() {
+        return service;
+    }
+
+    public void setService(ServiceTemplate service) {
+        this.service = service;
+    }
+
+    @Description("Template for Kafka headless service.")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public ServiceTemplate getHeadlessService() {
```

Comment:
I thought we were using the "broker service" and "bootstrap server" terminology?

Reply:
I though we agreed not to in the last comment in #817. But if you want I can change it. Its anyway in two separate classes. Not sure people would know what it means and what is the difference ... but they probably won't know either with service and headlessService so I'm fine to change it.